### PR TITLE
[1LP][RFR]Use tag object only updates

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
Update Taggable to work only with Tag object
{{pytest: -k 'edit_tag or test_drift_analysis or test_host_drift_analysis' --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 6e5a1c1ddecb067b06c6011db0026a4a40bb35ae:

`cfme/tests/ansible/test_embedded_ansible_actions.py`:
- :bangbang: Line 148:21: **E999** *SyntaxError: invalid syntax*

`cfme/tests/cloud_infra_common/test_vm_instance_analysis.py`:
- :bangbang: Line 711:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 718:13: **E131** *continuation line unaligned for hanging indent*

`cfme/tests/control/test_actions.py`:
- :bangbang: Line 789:20: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 796:21: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 876:28: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 884:32: **E127** *continuation line over-indented for visual indent*

`cfme/tests/infrastructure/test_host_drift_analysis.py`:
- :bangbang: Line 123:13: **E131** *continuation line unaligned for hanging indent*
- :bangbang: Line 130:13: **E131** *continuation line unaligned for hanging indent*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ffe1d754d0ee4d04c6d0e812080c72747048a680:

`cfme/common/__init__.py`:
- :red_circle: Line 346:56: **F821** *undefined name 'op'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by rlbabyuk** [comment]
PRT failures are not related to the changes

File: cfme/common/__init__.py [hunk 1]
```diff
@@ -261,72 +262,68 @@ class Taggable(object):
...
-        if details:
+        if details and not entities:
             view = navigate_to(self, 'EditTagsFromDetails')
-        else:
-            view = navigate_to(self, 'EditTags')
-        if isinstance(tag, Tag):
-            category = tag.category.display_name
-            tag = tag.display_name
+        elif entities or not details:
+            view = navigate_to(self, 'EditTags', *entities)
```
Comment (jamesooden) [inline cfme/common/__init__.py]: **QUESTION:** What happens here if entities is None?   The check above could allow us to get here if there are no details and entities is None if I understand correctly?   Is that what you really want?

File: cfme/common/__init__.py [hunk 2]
```diff
@@ -261,72 +262,68 @@ class Taggable(object):
...
```
Comment (jamesooden) [inline cfme/common/__init__.py]: **QUESTION:** Same as 279

File: cfme/common/__init__.py [hunk 3]
```diff
@@ -338,15 +335,19 @@ def remove_tags(self, tags):
...
-                self.remove_tag(category=category, tag=tag)
-        elif isinstance(tags, (list, tuple)):
-            for tag in tags:
-                self.remove_tag(tag=tag)
+        for tag in tags:
+            self.remove_tag(tag=tag)
+
+    def _set_random_tag(self, view):
+        random_cat = random.choice(view.form.tag_category.all_options).text
+        random_cat_cut = random_cat[:-1].strip() if random_cat[-1] == '*' else random_cat
```
Comment (jamesooden) [inline cfme/common/__init__.py]: **SUGGESTION:** Add comment to explain why * would be at the end of category name.

File: cfme/common/__init__.py [hunk 4]
```diff
@@ -338,15 +335,19 @@ def remove_tags(self, tags):
...
+        view.form.tag_category.fill(random_cat)  # In order to get the right tags list
```
Comment (jamesooden) [inline cfme/common/__init__.py]: **REQUIRED:** Add to comment to explain how this accomplishes getting the right tags list.

File: cfme/common/__init__.py (hunk 1 above)
Comment (rlbabyuk) [reply cfme/common/__init__.py]: I've added entities as an option to tag multiple items, but we can do it only in list view so if the entity is None only one item will be tagged, the one that tagging function is called, the same if details is False, we tag via the list

Comment (rlbabyuk) [reply cfme/common/__init__.py]: So, in the end, I decided to move this changes out of this PR, as this should be only for collections, and will be done in separate PR

**Pull request merged**